jerry-core/ecma/base/ecma-helpers.c

```diff
@@ -421,15 +473,26 @@ ecma_find_internal_property (ecma_object_t *object_p, /**< object descriptor */
   JERRY_ASSERT (property_id != ECMA_INTERNAL_PROPERTY_PROTOTYPE
                 && property_id != ECMA_INTERNAL_PROPERTY_EXTENSIBLE);
 
-  for (ecma_property_t *property_p = ecma_get_property_list (object_p);
-       property_p != NULL;
-       property_p = ECMA_GET_POINTER (ecma_property_t, property_p->next_property_p))
+  ecma_property_header_t *prop_iter_p = ecma_get_property_list (object_p);
+
+  while (prop_iter_p != NULL)
   {
-    if ((property_p->flags & ECMA_PROPERTY_FLAG_INTERNAL)
-        && property_p->h.internal_property_type == property_id)
+    JERRY_ASSERT (ECMA_PROPERTY_IS_PROPERTY_PAIR (prop_iter_p));
+
+    if (ECMA_PROPERTY_GET_TYPE (&prop_iter_p->types[0]) == ECMA_PROPERTY_TYPE_INTERNAL
```

Comment:
I don't want to change performance critical paths. However, I am willing to add asserts to these places.

There is another thing. Pair means two. If we want to add more items to a property list, it will not be called pair anymore, and the code needs to be refactored anyway.